### PR TITLE
Add language_name and language_facet

### PR DESCRIPTION
**Pull request by corylown**

### CODE REVIEW

File: lib/ddr/models/indexing.rb [hunk 1]
```diff
@@ -184,6 +186,12 @@ def date_sort
       date.first
     end
 
+    def language_name
+      language.map do |lang|
+        Language.codes.include?(lang) ? Language.find_by_code(lang).to_s : lang
```
Comment (dchandekstark) [inline lib/ddr/models/indexing.rb]: Consider instead:

```ruby
Language.find_by_code(lang).to_s rescue lang
```

File: lib/ddr/models/language.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+require "active_resource"
+
+module Ddr::Models
+  class Language < ActiveResource::Base
+
+    self.site = ENV["DDR_AUX_API_URL"]
+
+    def self.call(obj)
```
Comment (dchandekstark) [inline lib/ddr/models/language.rb]: How does this relate to `Ddr::Models::Indexing#language_name`?

File: spec/models/language_spec.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          it "returns a Language instance" do
+            expect(described_class.call(obj)).to eq([language])
+          end
+        end
+        describe "and the language is not found" do
+          before {
+            allow(described_class).to receive(:find_by_code).with("cym")
+                                       .and_raise(ActiveResource::ResourceNotFound, "404")
+          }
+          it "raises an exception" do
```
Comment (dchandekstark) [inline spec/models/language_spec.rb]: Just want to make sure this is the desired behavior.

File: lib/ddr/models/indexing.rb (hunk 1 above)
Comment (corylown) [inline lib/ddr/models/indexing.rb]: Good call. I'll push up the change.

File: lib/ddr/models/language.rb (hunk 2 above)
Comment (corylown) [inline lib/ddr/models/language.rb]: It doesn't right now. Eventually I'd like the indexing behavior to enforce using the codes and ignore or raise an error when a value in language isn't a valid code, but we're not there yet. I'd considered making the behavior of self.call be equivalent to what's currently happening in `Ddr::Models::Indexing#language_name` but it seemed odd to return a mix of strings and Language instances from self.call. Happy to reconsider.

Comment (dchandekstark) [inline lib/ddr/models/language.rb]: That's fine, just checking.

File: spec/models/language_spec.rb (hunk 3 above)
Comment (dchandekstark) [inline spec/models/language_spec.rb]: You answered.

File: lib/ddr/models/indexing.rb (hunk 1 above)
Comment (corylown) [inline lib/ddr/models/indexing.rb]: This change is now in the PR.

**Pull request merged**